src/averageAges.js

```diff
@@ -14,12 +14,34 @@
  *
  * @return {number}
  */
+const SEX_MALE = 'm';
+const SEX_FEMALE = 'f';
+const CENTURY_VALUE = 100;
+
+const getAverageAge = (peopleArray) =>
+  peopleArray.reduce((acc, person, personIndex) => {
+    const { born, died } = person;
+    const age = died - born;
+
+    if (personIndex === peopleArray.length - 1) {
+      return (acc + age) / peopleArray.length;
+    }
+
+    return acc + age;
+  }, 0);
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filteredPeopleBySex = people
+    .filter(({ sex, died }) => {
+      const deathCentury = Math.ceil(died / CENTURY_VALUE);
+      const isCenturyRequired = century
+        ? sex === SEX_MALE && deathCentury === century
+        : sex === SEX_MALE;
+
+      return isCenturyRequired;
+    });
+
+  return getAverageAge(filteredPeopleBySex);
```

Comment:
Consider combining the condition for 'century' into the filter method directly, instead of creating a separate variable inside the filter callback. This can help simplify the code.